Change to try to ignore __key__ as it is redundant

danieljbruce opened the pull request:
This change matches the change shown in another PR, but the test cases won’t allow for the new change. There must be some edge cases that haven’t been considered here.

danieljbruce committed:
Change to try to ignore __key__ as it is redundant

This change matches the change shown in another PR, but the test cases won’t allow for the new change. There must be some edge cases that haven’t been considered here.

Guillaume-Sensome commented:
This PR is linked to #1024.  
I didn't find the time to dive into tests to make it pass with this change.

Thanks to anyone doing it.

vicb commented:
The test failures tell us that 

https://github.com/googleapis/nodejs-datastore/blob/main/src/entity.ts#L672-L675

is never taken.

Did some debugging, the root cause is that the `Key` in tests in a copy (i.e. not the same as) the lib `Key` due to the the way it's imported in test.

https://github.com/vicb/nodejs-datastore/commit/184f0ede1a897a5ae18f6d3d93bad5d7cec995ce proves it

danieljbruce committed:
Add a check for isDsKey

isDsKey currently does not check to see that the key is an entity.Key properly so we need to modify this function so that it does a proper check.

danieljbruce committed:
Revert "Add a check for isDsKey"

This reverts commit 4f0db50bb26e54ac2c9b60d53beec6993d244de9.

danieljbruce commented:
https://github.com/googleapis/nodejs-datastore/pull/1085/files was the change adopted to solve this problem.
